src/main/java/com/cloudbees/jenkins/support/impl/RunningBuilds.java

```diff
@@ -0,0 +1,52 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.PrefilteredPrintedContent;
+import com.cloudbees.jenkins.support.filter.ContentFilter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Extension
+public class RunningBuilds extends Component {
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @Override
+    @NonNull
+    public String getDisplayName() {
+        return "Running Builds";
+    }
+
+    @Override
+    public void addContents(@NonNull Container result) {
+        result.add(
+            new PrefilteredPrintedContent("running-builds.txt") {
+                @Override
+                protected void printTo(PrintWriter out, ContentFilter filter) {
+                    Optional.ofNullable(Jenkins.getInstanceOrNull())
+                        .ifPresent(jenkins -> Arrays.stream(jenkins.getComputers())
+                            .flatMap(computer -> computer.getAllExecutors().stream())
+                            .collect(Collectors.toList())
+                            .forEach(executor -> Optional.ofNullable(executor.getCurrentExecutable())
+                                .filter(executable -> executable.getParent() == executable.getParent().getOwnerTask())
```

Comment:
Is it correct filtering before asking if the executable is present?

Reply:
Yes. See https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#filter-java.util.function.Predicate-